lib/mayaUsd/ufe/UsdAttribute.cpp

```diff
@@ -192,10 +194,74 @@ void setUsdAttributeVectorFromUfe(
     const PXR_NS::UsdTimeCode& time)
 {
     T vec;
-    vec.Set(value.x(), value.y(), value.z());
+    std::copy(value.vector.data(), value.vector.data() + value.vector.size(), vec.data());
+    setUsdAttr<T>(attr, vec);
+}
+
+template <typename T, typename U>
+U getUsdAttributeColorAsUfe(const PXR_NS::UsdAttribute& attr, const PXR_NS::UsdTimeCode& time)
+{
```

Comment:
But since we are accessing the inner class, we must create a specialization for colors because the name of the inner class is different.